ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -642,7 +643,7 @@ inline MatmulProgramConfig generate_matmul_program_config(
     const MemoryConfig& mem_config,
     const std::optional<const DeviceComputeKernelConfig> compute_kernel_config,
     const std::optional<const CoreCoord> user_core_coord,
-    const std::optional<const UnaryWithParam> user_fused_activation,
+    const std::optional<UnaryWithParam> user_fused_activation,
```

Comment:
Why remove the const?

Reply:
@bbradelTT In Matmul, both `std::optional<UnaryWithParam>` and ` const std::optional<const UnaryWithParam>` are used at different places. and using <const UnaryWithParam> throws an error.

Reply:
I think they should all be const. I'll need to clean that up afterwards.

Reply:
@KalaivaniMCW are you able to leave this as non-const for now? If not, what issues are you running into for the matmul code (@eyonland mentioned that you were running into some issues)?

Reply:
@bbradelTT I am able to use std::optional<UnaryWithParam> without any issues.